docs/design/2018-10-24-view-support.md

```diff
@@ -5,65 +5,120 @@
-  1.6 Store the view definition into following tables
-     * `INFORMATION_SCHEMA.VIEW_COLUMN_USAGE` lists one row for each column in a view including the base table of the column where possible 
-     * `INFORMATION_SCHEMA.VIEW_TABLE_USAGE` lists one row for each table used in a view 
-     * `INFORMATION_SCHEMA.VIEWS` lists one row for each view
-      
-2. Drop a view  
-  Implement DROP VIEW grammar, and delete the existing view tableinfo object.  
-3. Select from a view  
-  3.1 In function `func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error)`, if `tn *ast.TableName` is a view, then we build a select logicalplan from view's view_select string.   
-4. Insert/Update into a view  
-  I do not have any idea now, and will implement later.   
-5. Describe a view  
-  Generate the view select logicalplan. If any error occurs, return the error to parser. Otherwise, return `SELECT` cause's column info.
+1. Create VIEW  
+   This proposal only support following grammar to create view:
+   ```
+    CREATE
+        [OR REPLACE]
+        [ALGORITHM = {UNDEFINED | MERGE | TEMPTABLE}]
+        [DEFINER = { user | CURRENT_USER }]
+        [SQL SECURITY { DEFINER | INVOKER }]
+        VIEW view_name [(column_list)]
+        AS select_statement
+        [WITH [CASCADED | LOCAL] CHECK OPTION]
+    ```
+    1. Parse the create view statement and build a logical plan for select cause part. If any grammar error occurs, return errors to parser.   
+    2. Examine view definer's privileges. Definer should own both `CREATE_VIEW_PRIV` privilege and base table's `SELECT` privilege. We will resue `CREATE_PRIV` privilege when implement `CREATE VIEW` feature and will support `CREATE_VIEW_PRIV` check later.  
+    3. Examine create view statement, If ViewFieldList cause part is empty, then we should generate view column names from SelectStmt cause. Otherwise check len(ViewFieldList) == len(Columns from SelectStmt). And then we save column names to `TableInfo.Columns` .
+2. DROP VIEW  
+  Implement `DROP VIEW` grammar, and delete the existing view tableinfo object. This function should reuse `DROP TABLE` code logical
+3. SELECT FROM VIEW  
+    In function `func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error)`, if `tn *ast.TableName` is a view, then we build a select `LogicalPlan` from view's `SelectStmt` string. But this solution meet a problem,here is the example:  
+    ```mysql
+       create table t(a int,b int);
+       create view v like select * from t;
+       select * from t;
+    ```
+    Once we query from view `v`, database will rewrite view's `SelectStmt` from `select * from t` into **`select a as a,b as b from t`**
+    
+    ```mysql
+       drop table t;
+       create table t(c int,d int);
+    ```
+    If we rebuild table `v` from sql above and query from view `v` again, database will rewrite view's `SelectStmt` from `select * from t` into **`select c as c,d as d from t`**
```

Comment:
change line 92~93 to:
Executing `select * from v` now is equivalent to `select c as c, d as d from t`.The result of this query will be incompatible with the character of VIEW that a VIEW should be "frozen" after defined.